src/compile/c/excel_to_c_runtime.c

```diff
@@ -2359,11 +2359,18 @@ static ExcelValue hlookup(ExcelValue lookup_value_v,ExcelValue lookup_table_v, E
     for(i=0; i< columns; i++) {
       possible_match_v = array[i];
       if(lookup_value_v.type != possible_match_v.type) continue;
-      if(more_than(possible_match_v,lookup_value_v).number == true) {
-        if(i == 0) return NA;
-        return array[((((int) row_number_v.number)-1)*columns)+(i-1)];
-      } else {
-        last_good_match = i;
+      if(lookup_value_v.type == ExcelString && possible_match_v.type == ExcelString)
```

Comment:
@andreasDalsgaard do you have an example spreadsheet where this is needed? I've been writing some tests of HLOOKUP and I haven't yet managed to come up with one that matches Excel, but needs this.

Reply:
Sorry, for the long delay in responding. This was done as part of a consulting task and released with permission from the client. However, the client owns the spreadsheets and cannot be shared. I also don't have access to the spreadsheets files any more. I will try ask them if we can provide an example.